obs_maven/repo.py

```diff
@@ -27,20 +27,31 @@
 
 
 class Repo:
-    def __init__(self, uri, project, repository):
-        self.uri = uri
-        self.project = project.replace(":", ":/")
-        self.repository = repository
+    def __init__(self, base_url, project, repository, custom_url=None):
+        self.base_url = base_url
+        self.custom_url = custom_url
+        if not custom_url:
+            if project is not None and repository is not None:
```

Comment:
Same here tho. :wink: You are not really checking the type when creating it, i.e. `{name: Repo(self.url, data.get("project")...` — unless you are _really-really_ sure that `data['project']` is either always `None` or a string. Because `data.get(...)` will return `None` in two cases:

- No such key as "project"
- Key "project" contains `None`

I would do this:

```python
if not custom_url:
    if not project:
        raise ValueError("Project not defined")
    elif not repository:
        raise ValueError("Repository not defined")
    self.project = project.replace(":", ":/")
    self.repository = repository
```